pkg/otlp/metrics/metrics_remapping.go

```diff
@@ -69,7 +69,12 @@ func remapSystemMetrics(all pmetric.MetricSlice, m pmetric.Metric) {
 	case "system.filesystem.utilization":
 		copyMetric(all, m, "system.disk.in_use", 1)
 	}
-	// process.* and system.* metrics need to be prepended with the otel.* namespace
+	if strings.HasPrefix(name, "system.") {
+		// we keep the original metric for system.* metrics
```

Comment:
I looked again at the old code, I think the behavior is still different: we prepend `otel.` to the original names, but we keep the remapped names with no `otel.` prefix. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/23445/files#diff-59a3a2cae4395f15cad4800806f9dc7680aed0f303dd896c0ca8663e0fd0ac35L115-L129

For example, in the old code:
- if we have `"system.filesystem.utilization"`, first we remap it to `"system.disk.in_use"` and append the new one to slice
- then we prepend `otel.` to `"system.filesystem.utilization"` while 
`"system.disk.in_use"` is left unchanged
- in the end we got `"otel.system.filesystem.utilization"` and `"system.disk.in_use"`

In short, we prepend `otel.` to the original metrics, but don't prepend it to the remapped metrics.

Sorry for the confusion, I was also trying to understand the old behavior.

Reply:
I think you are misunderstanding the current PR. It works exactly like you expect (in your "old code" example above). I have simplified the tests for you to make it easier for you to understand. Feel free to set up a call if it's still not clear.

Reply:
If I'm not mistaken: using the same example, now after `remapMetrics` we're getting 3 metrics: `"otel.system.filesystem.utilization"`, `"system.disk.in_use"` and `"system.filesystem.utilization"`?

If so, the last one `"system.filesystem.utilization"` shouldn't be there